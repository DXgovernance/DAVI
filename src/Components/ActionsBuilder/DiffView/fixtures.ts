export const mockCodeOriginal = `
  {
    "colors": {
      "text": "red",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "lightGrey": "#BDC0C7",
        "grey": "orange"
      },
    }
  }
`;

export const mockCodeUpdated = `
  {
    "colors": {
      "text": "#fff",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "grey": "#A1A6B0",
        "lightGrey": "#BDC0C7"
      },
    }
  }
`;
