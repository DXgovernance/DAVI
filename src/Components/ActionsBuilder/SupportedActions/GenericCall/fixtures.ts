import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'Components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const genericDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.GENERIC_CALL,
  function: ERC20Contract.getFunction('increaseAllowance'),
  to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  value: BigNumber.from(0),
  args: {
    param1: BigNumber.from(10000),
    param2: BigNumber.from(10000),
    param3: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
