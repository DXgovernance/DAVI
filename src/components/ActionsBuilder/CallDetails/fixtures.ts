import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20.abi);

export const emptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const decodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0x95a223299319022a842d0dfe4851c145a2f615b9',
  value: BigNumber.from(0),
  args: {
    _to: '0x0000000000000000000000000000000000000000',
    _value: BigNumber.from(1000),
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  callType: SupportedAction.GENERIC_CALL,
  from: '0x0000000000000000000000000000000000000000',
  to: '0x0000000000000000000000000000000000000000',
  value: BigNumber.from(0),
  function: null,
  args: {},
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
