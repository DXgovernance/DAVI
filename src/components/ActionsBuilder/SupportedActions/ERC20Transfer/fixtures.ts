import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const erc20TransferEmptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const erc20TransferDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0xfFb1cd0F95368DDd06D556161c5D3d9f0f4Fe6d2',
  value: BigNumber.from(0),
  args: {
    _to: '0xaF8eB8C3A5d9d900AA0B98e3Df0bcC17d3C5F698',
    _value: BigNumber.from('5000000000000000000'),
  },
};
