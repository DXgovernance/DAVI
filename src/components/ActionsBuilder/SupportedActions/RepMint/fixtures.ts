import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import { BigNumber, utils } from 'ethers';
import ERC20SnapshotRep from 'contracts/ERC20SnapshotRep.json';

const ERC20SnapshotRepContract = new utils.Interface(ERC20SnapshotRep.abi);

export const repMintEmptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.REP_MINT,
  function: ERC20SnapshotRepContract.getFunction('mint'),
  to: '',
  value: BigNumber.from(0),
  args: {
    to: '',
    amount: BigNumber.from(0),
  },
};

export const repMintDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.REP_MINT,
  function: ERC20SnapshotRepContract.getFunction('mint'),
  to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
  value: BigNumber.from(0),
  args: {
    to: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
    amount: BigNumber.from(10000),
  },
};
