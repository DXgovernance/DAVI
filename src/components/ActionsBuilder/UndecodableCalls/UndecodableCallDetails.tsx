import { MetadataTag } from '../SupportedActions/common/infoLine';
import { DetailRow } from '../SupportedActions/common/Summary.styled';
import { Call } from '../types';
import { BigNumber } from 'ethers';
import { Box } from 'components/primitives/Layout/Box';
import { BlockExplorerLink } from 'components/primitives/Links';
import styled, { useTheme } from 'styled-components';

const ParamTitleRow = styled(DetailRow)`
  margin-bottom: 0.75rem;
`;

const ParamTitleTag = styled(MetadataTag)`
  display: inline-block;
  padding: 0.375rem 0.5rem;
  color: ${({ color }) => color};
`;

const ActionParamRow = styled(Box)`
  margin-bottom: 1.5rem;
`;

const ParamDetail = styled(Box)`
  color: ${({ theme }) => theme.colors.grey};
  overflow-wrap: break-word;
`;

const UndecodableCallDetails: React.FC<{ call: Call }> = ({ call }) => {
  const theme = useTheme();

  function renderByCallData(key: string, value: any) {
    if (!key || !value) return null;

    if (key === 'to' || key === 'from') {
      return <BlockExplorerLink address={value} unstyled avatarSize={16} />;
    }

    if (key === 'value') {
      return <ParamDetail>{BigNumber.from(value).toString()}</ParamDetail>;
    }

    return <ParamDetail>{value}</ParamDetail>;
  }

  const renderedCallDetails: Call = {
    from: call.from,
    to: call.to,
    data: call.data,
    value: call.value,
  };

  return (
    <>
      {Object.entries(renderedCallDetails)?.map(([key, value], index) => (
        <ActionParamRow key={index}>
          <ParamTitleRow>
            <ParamTitleTag color={theme?.colors?.params?.[index]}>
              {key}
            </ParamTitleTag>
          </ParamTitleRow>

          {renderByCallData(key, value)}
        </ActionParamRow>
      ))}
    </>
  );
};

export default UndecodableCallDetails;
