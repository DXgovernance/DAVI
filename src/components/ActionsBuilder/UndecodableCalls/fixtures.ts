import { BigNumber } from 'ethers';
import { Call, SupportedAction } from '../types';

export const MOCK_ADDRESS = '0x0000000000000000000000000000000000000000';
export const MOCK_ENS_NAME = 'name.eth';

export const callMock: Call = {
  from: '0x0000000000000000000000000000000000000000',
  to: '',
  data: '0x2e1a7d4d00000000000000000000000000000000000000000000000001aa0b0f33c98800',
  value: BigNumber.from('1000000000000000000'),
};

export const callMockWithApproval: Call = {
  ...callMock,
  approval: {
    callType: SupportedAction.GENERIC_CALL,
    from: '0x0000000000000000000000000000000000000000',
    to: '0x0000000000000000000000000000000000000000',
    value: BigNumber.from(0),
    function: null,
    args: {},
    amount: BigNumber.from('10000'),
    token: '0x0000000000000000000000000000000000000000',
  },
};
