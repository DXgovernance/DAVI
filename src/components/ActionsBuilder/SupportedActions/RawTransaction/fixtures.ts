import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import { BigNumber } from 'ethers';
import { MOCK_ADDRESS } from 'hooks/Guilds/ens/fixtures';

export const rawDataCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.RAW_TRANSACTION,
  function: null,
  to: MOCK_ADDRESS,
  value: BigNumber.from('1000000000000000000'),
  args: null,
  optionalProps: {
    data: '0x01',
  },
};
