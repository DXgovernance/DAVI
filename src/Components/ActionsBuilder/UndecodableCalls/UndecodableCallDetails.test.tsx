import { render } from 'utils/tests';
import { callMock, callMockWithApproval } from './fixtures';
import UndecodableCallDetails from './UndecodableCallDetails';

describe('UndecodableCallDetails', () => {
  it('Should match snapshot', () => {
    const { container } = render(<UndecodableCallDetails call={callMock} />);
    expect(container).toMatchSnapshot();
  });
  it('Should match snapshot with approval call', () => {
    const { container } = render(
      <UndecodableCallDetails call={callMockWithApproval} />
    );
    expect(container).toMatchSnapshot();
  });
});
