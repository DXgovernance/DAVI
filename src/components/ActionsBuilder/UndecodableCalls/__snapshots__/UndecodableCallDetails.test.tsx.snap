// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UndecodableCallDetails Should match snapshot 1`] = `
.c4 {
  padding: 0.125rem 0.375rem;
  background-color: #303338;
  border-radius: 10px;
}

.c0 {
  box-sizing: 'border-box';
  min-width: 0;
  margin: 0;
  padding: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  color: #A1A6B0;
  overflow-wrap: break-word;
}

.c7 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
}

.c7:focus,
.c7:hover,
.c7:visited,
.c7:link,
.c7:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-bottom: 0.75rem;
}

.c5 {
  display: inline-block;
  padding: 0.375rem 0.5rem;
  color: #9575CD;
}

.c9 {
  display: inline-block;
  padding: 0.375rem 0.5rem;
  color: #F06292;
}

.c10 {
  display: inline-block;
  padding: 0.375rem 0.5rem;
  color: #4DB6AC;
}

.c12 {
  display: inline-block;
  padding: 0.375rem 0.5rem;
  color: #64B5F6;
}

.c1 {
  margin-bottom: 1.5rem;
}

.c11 {
  color: #A1A6B0;
  overflow-wrap: break-word;
}

<div>
   
  <div
    class="c0 c1"
  >
    <div
      class="c0 c2 c3"
    >
      <span
        class="c4 c5"
        color="#9575CD"
      >
        from
      </span>
    </div>
    <div
      class="c6"
    >
      <a
        class="c7"
        rel="noopener"
        target="_blank"
      >
        <div
          class="c0 c8"
        >
          0x0000...0000
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
            <polyline
              points="15 3 21 3 21 9"
            />
            <line
              x1="10"
              x2="21"
              y1="14"
              y2="3"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div
      class="c0 c2 c3"
    >
      <span
        class="c4 c9"
        color="#F06292"
      >
        to
      </span>
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div
      class="c0 c2 c3"
    >
      <span
        class="c4 c10"
        color="#4DB6AC"
      >
        data
      </span>
    </div>
    <div
      class="c0 c11"
    >
      0x2e1a7d4d00000000000000000000000000000000000000000000000001aa0b0f33c98800
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div
      class="c0 c2 c3"
    >
      <span
        class="c4 c12"
        color="#64B5F6"
      >
        value
      </span>
    </div>
    <div
      class="c0 c11"
    >
      1000000000000000000
    </div>
  </div>
   
</div>
`;

exports[`UndecodableCallDetails Should match snapshot with approval call 1`] = `
.c4 {
  padding: 0.125rem 0.375rem;
  background-color: #303338;
  border-radius: 10px;
}

.c0 {
  box-sizing: 'border-box';
  min-width: 0;
  margin: 0;
  padding: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 0.5rem;
  margin-bottom: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  color: #A1A6B0;
  overflow-wrap: break-word;
}

.c7 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
}

.c7:focus,
.c7:hover,
.c7:visited,
.c7:link,
.c7:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  margin-bottom: 0.75rem;
}

.c5 {
  display: inline-block;
  padding: 0.375rem 0.5rem;
  color: #9575CD;
}

.c9 {
  display: inline-block;
  padding: 0.375rem 0.5rem;
  color: #F06292;
}

.c10 {
  display: inline-block;
  padding: 0.375rem 0.5rem;
  color: #4DB6AC;
}

.c12 {
  display: inline-block;
  padding: 0.375rem 0.5rem;
  color: #64B5F6;
}

.c1 {
  margin-bottom: 1.5rem;
}

.c11 {
  color: #A1A6B0;
  overflow-wrap: break-word;
}

<div>
   
  <div
    class="c0 c1"
  >
    <div
      class="c0 c2 c3"
    >
      <span
        class="c4 c5"
        color="#9575CD"
      >
        from
      </span>
    </div>
    <div
      class="c6"
    >
      <a
        class="c7"
        rel="noopener"
        target="_blank"
      >
        <div
          class="c0 c8"
        >
          0x0000...0000
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
            <polyline
              points="15 3 21 3 21 9"
            />
            <line
              x1="10"
              x2="21"
              y1="14"
              y2="3"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div
      class="c0 c2 c3"
    >
      <span
        class="c4 c9"
        color="#F06292"
      >
        to
      </span>
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div
      class="c0 c2 c3"
    >
      <span
        class="c4 c10"
        color="#4DB6AC"
      >
        data
      </span>
    </div>
    <div
      class="c0 c11"
    >
      0x2e1a7d4d00000000000000000000000000000000000000000000000001aa0b0f33c98800
    </div>
  </div>
  <div
    class="c0 c1"
  >
    <div
      class="c0 c2 c3"
    >
      <span
        class="c4 c12"
        color="#64B5F6"
      >
        value
      </span>
    </div>
    <div
      class="c0 c11"
    >
      1000000000000000000
    </div>
  </div>
   
</div>
`;
